src/test/__tests__/utils.int.test.ts

```diff
@@ -11,4 +11,55 @@ describe('test/utils.ts', () => {
 			expect(isoTimestampPattern.test('2022-10-27')).toBe(false);
 		});
 	});
+
+	describe('generateNextWithAssertions', () => {
```

Comment:
Agreed that this a proactive for handling the introduction of more involved/complex middleware, and code is functional!